notebooks/05-operations/01-druid-logs-configuration.ipynb

```diff
@@ -0,0 +1,511 @@
+    "| Coordinator / Overlord | __org.apache.druid.server.coordinator.rules__<br>Lots of information about how the retention rules are actually being applied (or not!). |\n",
+    "| Coordinator / Overlord | __org.apache.druid.server.coordinator.duty.BalanceSegments__<br>Here you can see what Druid decides to do when balancing needs to be carried out – e.g. a server is lost or added. |\n",
+    "| Historical | __org.apache.druid.server.coordination.BatchDataSegmentAnnouncer__<br>You can see individual segments being announced as available for query by each historical server as it loads them. |\n",
+    "| Historical | __org.apache.druid.server.coordination.SegmentLoadDropHandler__<br>As well as seeing how the historical checks its local segment cache on startup, you can watch along as the Historical picks up the instructions from the coordinator and then does something about them. When there are ERRORs like “Failed to load segment for dataSource” you get traces about what the process was trying to do – quite often something pointing to an error with its connection to deep storage. |"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "57b38097-3c09-4fc2-b163-433a46e3b32f",
+   "metadata": {},
+   "source": [
+    "In your `multitail` window, press `q` to quit and return to the terminal."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "df2f17b3-f359-4df8-a682-7398e7d2adf8",
+   "metadata": {},
+   "source": [
+    "### Log retention\n",
+    "\n",
+    "Since \"FileAppender\" is a [RollingRandomAccessFileAppender](https://logging.apache.org/log4j/log4j-2.4/manual/appenders.html#RollingRandomAccessFileAppender) you can adjust the `DefaultRolloverStrategy` to control retention of logs by adjusting the `Delete` section.\n",
+    "\n",
+    "`IfFileName` and `IfLastModified` are used in conjunction to remove any files from the log folder that match the rules. The default is to remove matching files older than two months, based on the date last modified.\n",
+    "\n",
+    "Run the following command to change the trigger policy to a [duration](https://logging.apache.org/log4j/2.x/javadoc/log4j-core/org/apache/logging/log4j/core/appender/rolling/action/Duration.html#parseCharSequence) of \"PT2M\". As a consequence, any files matching the `IfFileName` filter older than 2 minutes will be deleted automatically by Log4J.\n",
+    "\n",
+    "Looking back at your previous `ls -l` results, run this command and repeat to see how the `Delete` rule is applied.\n",
+    "\n",
+    "```bash\n",
+    "sed -i '' 's/IfLastModified age=\"7d\"/IfLastModified age=\"PT2M\"/' \\\n",
+    "  ~/learn-druid-local/apache-druid/conf/druid/auto/_common/log4j2.xml\n",
+    "```\n",
+    "\n",
+    "Notice how _all_ files are removed, not just rollover logs.\n",
+    "\n",
+    "Run this command to change the retention policy to one day.\n",
+    "\n",
+    "```bash\n",
+    "sed -i '' 's/IfLastModified age=\"PT2M\"/IfLastModified age=\"P1D\"/' \\\n",
```

Comment:
For this one you may need to wait up to a minute to see that more files are being retained, perhaps add a comment that says so.